packages/contracts-bedrock/scripts/go-ffi/bundle.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+)
+
+type Transaction struct {
+	To             string  `json:"to"`
+	Value          string  `json:"value"`
+	Data           *string `json:"data"`
+	ContractMethod struct {
+		Inputs []struct {
+			Name string `json:"name"`
+			Type string `json:"type"`
+		} `json:"inputs"`
+		Name string `json:"name"`
+	} `json:"contractMethod"`
+	ContractInputsValues map[string]string `json:"contractInputsValues"`
+}
+
+type Bundle struct {
+	Transactions []Transaction `json:"transactions"`
+}
+
+// EncodedTransaction represents the ABI-encoded transaction
+type EncodedTransaction struct {
+	Target   common.Address
+	Value    *big.Int
+	Calldata []byte
+}
+
+func EncodeBundleTransactions() {
+	if len(os.Args) < 3 {
+		fmt.Println("Usage: go run main.go EncodeBundleTransactions <path_to_json_file>")
+		os.Exit(1)
+	}
+
+	jsonFilePath := os.Args[2]
+
+	// Read the JSON file
+	jsonData, err := os.ReadFile(jsonFilePath)
+	if err != nil {
+		fmt.Printf("Error reading JSON file: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Parse the JSON data
+	var bundle Bundle
+	err = json.Unmarshal(jsonData, &bundle)
+	if err != nil {
+		fmt.Printf("Error parsing JSON data: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Encode the transactions
+	encodedTransactions := make([]EncodedTransaction, len(bundle.Transactions))
+	for i, tx := range bundle.Transactions {
+		target := common.HexToAddress(tx.To)
+		value, _ := new(big.Int).SetString(tx.Value, 10)
+
+		if tx.Data != nil {
+			fmt.Println("Data MUST be null")
+			os.Exit(1)
+		}
+
+		calldata, err := encodeMethodCall(tx.ContractMethod, tx.ContractInputsValues)
+		if err != nil {
+			fmt.Printf("Error encoding method call: %v\n", err)
+			os.Exit(1)
+		}
+		encodedTransactions[i] = EncodedTransaction{
+			Target:   target,
+			Value:    value,
+			Calldata: calldata,
+		}
+	}
+
+	// Define the array type for EncodedTransaction
+	arrayType, err := abi.NewType("tuple[]", "", []abi.ArgumentMarshaling{
```

Comment:
I recommend checking out https://github.com/lmittmann/w3 for these small sorts of tasks, its a nicer syntax wrapper on top of this